Make mr_proper fails when the directory does not have the correct name

When using

`make mr_proper arduino-cli-download safe-install`

After cloning the repo without renaming it `LampColorControler`, will fail in obscure manner: 

```
 --- has-board-installed
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
Can't open sketch: main file missing from sketch: /home/epsilon/dev/Lamp-Da/Lamp-Da.ino
 --- check-compiler-cmd
/bin/bash: line 2: --version: command not found


ERROR: compiler "" seems to be inappropriate?

Troubleshooting:
 - detected compiler path is ""
 - detected compiler command is ""
 - output of " --version" below:

/bin/bash: line 11: --version: command not found
 - expected "Arm Embedded Processors" in compiler version string
 - packages like "libc6-dev-i386" or "lib32-glibc" may be needed locally for 32-bit support
 - packages like "gcc-arm-none-eabi" or "arm-none-eabi-gcc" may be needed locally for ARM support
 - you can also try "make mr_proper safe-install" (may take 1.4 gigabyte of space)

make[2]: *** [Makefile:152: check-compiler-cmd] Error 1
make[2]: Leaving directory '/home/epsilon/dev/Lamp-Da'
make[1]: *** [Makefile:172: /home/epsilon/dev/Lamp-Da/_build/.deps-ok] Error 2
make[1]: Leaving directory '/home/epsilon/dev/Lamp-Da'
make: *** [Makefile:269: safe-install] Error 2
```

The fix is renaming the directory (or just reading the instructions and cloning it correctly), but it would be better to output an helpful message reminding the user to do so :)

--- a/Makefile
+++ b/Makefile
@@ -21,8 +21,10 @@ ADAFRUIT_URL=https://adafruit.github.io/arduino-board-index/package_adafruit_ind
 CUSTOM_BOARD_URL=https://github.com/BaptisteHudyma/LampDa_nRF52_Arduino
 FULL_LAMP_TYPE=LMBD_LAMP_TYPE__$(shell echo ${LMBD_LAMP_TYPE} | tr '[:lower:]' '[:upper:]')
 
+FOLDER_NAME=$(shell basename $(SRC_DIR))
 FQBN=adafruit:nrf52:lampDa_nrf52840
-PROJECT_INO=LampColorControler.ino
+PROJECT_NAME=LampColorControler
+PROJECT_INO=$(PROJECT_NAME).ino
 COMPILER_CMD=$(shell $(ARDUINO_CLI) compile -b $(FQBN) --show-properties|grep compiler.cpp.cmd|cut -f2 -d=)
 COMPILER_PATH=$(shell $(ARDUINO_CLI) compile -b $(FQBN) --show-properties|grep compiler.path|cut -f2 -d=)
 
@@ -76,7 +78,20 @@ upload-simple:
 upload-indexable:
 	LMBD_LAMP_TYPE="indexable" make upload
 
-has-lamp-type:
+
+has-correct-folder-name:
+	@echo; echo " --- $@"
+	@[ $(FOLDER_NAME) == $(PROJECT_NAME) ] || \
+		(echo; echo \
+		 ; echo 'ERROR: The folder name should be $(PROJECT_NAME)' \
+		 ; echo \
+		 ; echo 'The repository should have been cloned using' \
+		 ; echo '    git clone https://github.com/BaptisteHudyma/Lamp-Da.git $(PROJECT_NAME)' \
+		 ; echo \
+		 ; echo \
+		 ; false)
+
+has-lamp-type: has-correct-folder-name
 	@echo; echo " --- $@"
 	@test ! -z "$$LMBD_LAMP_TYPE" || \
 		(echo; echo \
@@ -97,6 +112,7 @@ has-lamp-type:
 		 ; echo \
 		 ; echo \
 		 ; false)
+
 
 #
 # dependency checks
@@ -567,7 +583,7 @@ simulator: indexable-simulator
 
 remove: mr_proper
 
-mr_proper:
+mr_proper: has-correct-folder-name
 	@echo; echo " --- $@"
 	@(test ! -L venv && rm -rf venv) || true
 	rm -rf docs/html/*

